src/main/java/jenkins/scm/api/mixin/ChangeRequestSCMHead.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package jenkins.scm.api.mixin;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.SCMSource;
+import org.kohsuke.stapler.export.Exported;
+
+/**
+ * Mixin interface to identify {@link SCMHead} instances that correspond to a change request.
+ *
+ * @since 2.0
+ */
+public interface ChangeRequestSCMHead extends SCMHeadMixin {
+
+    /**
+     * Identifier of this change request.
+     * Expected to be unique among requests coming from a given {@link SCMSource}.
+     *
+     * @return an ID of some kind, such as a pull request number (decimal) or a Gerrit change ID
+     */
+    @Exported
+    @NonNull
+    String getId();
+
+    /**
+     * Branch to which this change would be merged or applied if it were accepted.
```

Comment:
I still do not see any compelling reason why this should be considered part of the identity of the head, when GitHub (the service, not our _current_ plugin) supports changing it on an existing PR.